src/network/networkd-ndisc.c

```diff
@@ -1175,15 +1239,31 @@ static int ndisc_router_handler(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return r;
 
-        r = ndisc_drop_outdated(link, timestamp_usec);
+        r = sd_ndisc_router_get_lifetime(rt, &router_lifetime_sec);
         if (r < 0)
-                return r;
+                return log_link_warning_errno(link, r, "Failed to get lifetime of RA message: %m");
 
-        r = ndisc_start_dhcp6_client(link, rt);
+        /* https://datatracker.ietf.org/doc/html/rfc4861
+         * Router Lifetime: A Lifetime of 0 indicates that the router is not a default router
+         * and SHOULD NOT appear on the default router list.
+         */
+        if (router_lifetime_sec == 0) {
+                r = ndisc_process_zero_lifetime(link, rt);
+                if (r < 0)
+                        log_link_warning_errno(link, r, "Failed to remove SLAAC route having lifetime = 0, ignoring: %m");
+
+                return 0;
+        }
+
+        r = ndisc_drop_outdated(link, timestamp_usec);
         if (r < 0)
                 return r;
 
         r = ndisc_router_process_default(link, rt);
+        if (r < 0)
+               return r;
+
+        r = ndisc_start_dhcp6_client(link, rt);
```

Comment:
Hmm. Please do not unnecessarily change the order of the functions.
1. drop the outdated configurations,
2. start DHCPv6 client,
3. process default.